amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -225,9 +225,15 @@ RecordsRetrieved getRecordsRetrieved() {
         }
 
         // This method is not thread-safe. You need to acquire a lock in the caller in order to execute this.
-        void executeEventAction(Subscriber<? super RecordsRetrieved> subscriber) {
+        void executeEventAction(Subscriber<? super RecordsRetrieved> subscriber, BlockingQueue<RecordsRetrievedContext> recordsDeliveryQueue) {
             recordsOrShutdownEvent.apply(recordsEvent -> subscriber.onNext(recordsEvent),
-                    shutdownEvent -> shutdownEvent.getSubscriptionShutdownAction().run());
+                    shutdownEvent -> handleShutdownEvent(shutdownEvent, recordsDeliveryQueue));
+        }
+
+        private void handleShutdownEvent(SubscriptionShutdownEvent shutdownEvent, BlockingQueue<RecordsRetrievedContext> recordsDeliveryQueue) {
+            //Removing the Shutdown Event from the queue before executing the Shutdown action.
+            recordsDeliveryQueue.poll();
```

Comment:
Let's add this to the ShutdownEvent runnable, rather than having a new method here.